File: test/exercism/ScrabbleScore.ts
const scrabblePoints: Record<string, number> = Object.freeze({
  a: 1, e: 1, i: 1, o: 1, u: 1, l: 1, n: 1, r: 1, s: 1, t: 1,
  d: 2, g: 2,
  b: 3, c: 3, m: 3, p: 3,
  f: 4, h: 4, v: 4, w: 4, y: 4,
  k: 5,
  j: 8, x: 8,
  q: 10, z: 10
})

function score(word?: string | null): number {
  const letterScore = (c: string) => scrabblePoints[c.toLowerCase()] ?? 0

  // @ts-ignore - false positive - works in https://www.typescriptlang.org/play
  return [...word ?? []].reduce((s, c) => s + letterScore(c), 0)
}

// unit tests
[
  { input: null, expected: 0 },
  { input: undefined, expected: 0 },
  { input: '', expected: 0 },
  { input: 'a', expected: 1 },
  { input: 'A', expected: 1 },
  { input: 'z', expected: 10 },
  { input: 'Zoo', expected: 12 },
  { input: 'street', expected: 6 },
  { input: 'abcdefghijklmnopqrstuvwxyz', expected: 87 },
].map(
  t => ({ ...t, actual: score(t.input) })
).forEach(
  t => console.log(
    `score(${t.input}) -> ${t.actual}`,
    t.actual === t.expected ? '✔' : `❌ expected: ${t.expected}`
  )
)

File: test/exercism/ScrabbleScore.kt
package exercism

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

/**
 * https://exercism.io/tracks/kotlin/exercises/scrabble-score
 *
 * Scrabble Score
 * [Easy]
 *
 * Given a word, compute the Scrabble score for that word.
 */
object ScrabbleScore {

    fun scoreWord(word: String) = ScrabbleWord(word).score
}

class ScrabbleWord(word: String) {

    val score by lazy {
        word.sumOf { it.scrabblePoints }
    }
}

private val Char.scrabblePoints
    get() = when (uppercaseChar()) {
        'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' -> 1
        'D', 'G' -> 2
        'B', 'C', 'M', 'P' -> 3
        'F', 'H', 'V', 'W', 'Y' -> 4
        'K' -> 5
        'J', 'X' -> 8
        'Q', 'Z' -> 10
        else -> 0
    }

/**
 * Unit tests
 */
class ScrabbleScoreTest {

    @ParameterizedTest
    @CsvSource(
        "a, 1",
        "A, 1",
        "f, 4",
        "at, 2",
        "zoo, 12",
        "street, 6",
        "quirky, 22",
        "OxyphenButazone, 41",
        "pinata, 8",
        ", 0",
        "abcdefghijklmnopqrstuvwxyz, 87"
    )
    fun test(input: String?, expectedOutput: Int) {
        assertEquals(expectedOutput, ScrabbleScore.scoreWord(input.orEmpty()))
    }
}
